src/testing/mocks/middleware/icache.ts

```diff
@@ -0,0 +1,59 @@
+/* tslint:disable:interface-name */
+import { create, invalidator } from '../../../core/vdom';
+import { MiddlewareResult } from '../../../core/interfaces';
+import { cache } from '../../../core/middleware/cache';
+import { icache } from '../../../core/middleware/icache';
+import Map from '../../../shim/Map';
+
+export function createICacheMock() {
+	const map = new Map<string, any>();
+	const factory = create({ cache, invalidator });
+	const mockICacheFactory = factory(({ id, middleware, properties, children }) => {
+		const { callback } = icache();
+		const icacheMiddleware = callback({
+			id,
+			middleware: { invalidator: middleware.invalidator, cache: middleware.cache },
+			properties,
+			children
+		});
+		const setter = icacheMiddleware.set;
+
+		icacheMiddleware.set = (key: any, value: any) => {
+			if (typeof value === 'function') {
+				value = value();
+				map.set(key, value);
```

Comment:
Do you need to set this in the exposed map if it's not a promise?

Reply:
Are you thinking the mock would fall back to the middleware if the `map` did not contain a key? It doesn't now, but if it did, then no; the `map` would only need to track the promises.

Reply:
I didn't know the reason why it would need to expose values that weren't a promise, the only reason we need to expose the promise is to await the resolution?

Reply:
I'm fine with that. Users can always expand the idea with their own mock if they have specific needs.

Reply:
Resolved with c45e07aab8ab50d633b5f548a2c5bede22ef2c48